### PR TITLE
RCPS-140: stud-info-mgt-app: Show password on forgot password page al…

### PR DESCRIPTION
**Pull request by RichardNeilM**
I've added a toggle button for users to view their password, resolving the issue.

### CODE REVIEW

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 1]
```diff
@@ -61,7 +62,7 @@ public class AddStudentController implements Initializable {
     @FXML
     private Button studentAddButton;
 
-    private StudentFacade studentFacade = new StudentFacadeImpl();
+    private StudentFacade studentFacade = new StudentFacadeImpl(new StudentDaoImpl());
```
Comment (RoCS2024) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java]: Daos should not be visible in UI

**Comment by RichardNeilM** [comment]
the "new update" commit is for the Dao not visible in the UI

the "RCPS-163: stud-info-mgt-app:ADDSTUDENT-display invalid input" commit is the new ticket that I pushed to this PR

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 2]
```diff
@@ -67,32 +67,101 @@ public class AddStudentController implements Initializable {
...
+        alert.setTitle(title);
+        alert.setHeaderText(null);
+        alert.setContentText(message);
+        alert.showAndWait();
+    }
+
+
+
+
+    private void showAlert(String message) {
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java:156]: does this method have any usage or a duplicate code?

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 3]
```diff
@@ -67,32 +67,101 @@ public class AddStudentController implements Initializable {
...
-            ex.printStackTrace();;
+        } catch (Exception ex) {
+            ex.printStackTrace();
         }
-        Stage stage = (Stage) ((Node)event.getSource()).getScene().getWindow();
+        Stage stage = (Stage) ((Node) event.getSource()).getScene().getWindow();
         stage.close();
     }
 
+    private void showAlert(String title, String message) {
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java:145]: if exit and ok are clicked in the alert, return to stage.

File: src/main/java/com/student/user/app/controllers/item/AddStudentController.java [hunk 4]
```diff
@@ -67,32 +67,101 @@ public class AddStudentController implements Initializable {
...
+        if (!firstName.getText().matches(genericRegex)) {
+            return "Invalid input for First Name.";
+        }
+        if (!middleName.getText().matches(genericRegex)) {
+            return "Invalid input for Middle Name.";
+        }
+        if (!religion.getText().matches(genericRegex)) {
+            return "Invalid input for Religion.";
+        }
+        if (!contactNo.getText().matches(genericRegex)) {
```
Comment (zhairanicolepacheco) [inline src/main/java/com/student/user/app/controllers/item/AddStudentController.java:102]: invalid validation for contact  number

**Pull request closed without merging**